### DISCUSSION TITLE
wrong-type-argument characterp nil

### DISCUSSION THREAD

**Discussion by fedeizzo** [minad/jinx#175] [Q&A] [answered]
Hello 👋 .

I installed jinx and I am trying to run it for the first time but I am encountering a problem. I attach the output with `debug-on-error` set to `t`.

```
Debugger entered--Lisp error: (wrong-type-argument characterp nil)
  jinx--correct-suggestions("here")
  #f(compiled-function () #<bytecode -0x12196504f82acdc5>)()
  jinx--correct-highlight(#<overlay from 226 to 230 in jinx.el<packages>> #f(compiled-function () #<bytecode -0x12196504f82acdc5>))
  jinx--correct-overlay(#<overlay from 226 to 230 in jinx.el<packages>>)
  jinx-correct-nearest()
  jinx-correct(nil)
  funcall-interactively(jinx-correct nil)
  command-execute(jinx-correct)
```

with this buffer content

```elisp
;;; Code:
(use-package jinx
  :hook (emacs-startup . global-jinx-mode)
  :bind (("M-$" . jinx-correct)
         ("C-M-$" . jinx-languages)))
;;; jinx.el ends here
```

I am running nix+home manager on MacOS and I am using emacs-overlay with the nix definition contained in `nixpkgs`, version 24.05.

**Comment by minad** [comment]
Hi, thanks for the report. Unfortunately I cannot reproduce this problem. Do you use the newest version of Jinx? Do you see the issue with emacs -Q? Which Emacs version do you use?

**Reply by fedeizzo** [reply]
Jinx version: 20240415.423
Emacs version: GNU Emacs 29.3
Error with emacs -Q: yes

I am compiling to bytecode (as you can see from the log) all the packages, could it be related?

**Reply by minad** [reply]
This doesn't seem like the newest version (see the exact commit or the package version number). But anyway, I am afraid I cannot help you without reproducing the problem myself. Maybe you can try debugging this and see where the problem lies.

**Comment by svaante** [comment]
First of thank you for your work @minad!

I am able to produce the same `*Backtrace*` buffer as @fedeizzo.

Set `jinx--dicts` to `nil`.

On system `jinx--dicts` was `nil`ed due to `jinx-languages` being messed up due to `current-locale-environment` being set to  `"UTF-8"` (which is *not* an `jinx` issue but an unrelated issue)

Maybe it would be an improvement if `(message "Jinx: No dictionaries available for %S" jinx-languages)` in `jinx--load-dicts` was promoted to an `warn`ing.

**Reply by minad** [reply, marked as answer]
Thanks. Fixed by https://github.com/minad/jinx/commit/81d0a092b3ca4c5036af45c9411ebbe66eb123ea.
